### PR TITLE
ci: replace travis with actions, enable renovate

### PR DESCRIPTION
**Pull request by gr2m**
### Todos

- [x] delete `.travis.yml`
- [x] create `.github/workflows/{test,release}.yml`
- [x] remove `package-lock.json` from `.gitignore`
- [x] delete `.npmrc` and `.yarnrc`
- [x] README
	- [x] remove travis/codecov badges
	- [x] add actions badge instead (replace `REPO NAME HERE` ;): `[Build Status](https://github.com/semantic-release/[REPO NAME HERE]/actions?query=workflow%3ATest+branch%3Amaster)`
- [x] create `package-json.lock`
- [x] add to package.json:
	- [x] `test:ci` script which doesn't run `lint` pre script
	- [x] renovate config 

		```
		  "renovate": {
		    "extends": [
		      "github>semantic-release/.github"
		    ]
		  }
		```
- [x] update branch protection: remove required status by Travis and Codecov, add required status for "test" (wait for first build of PR)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +0,0 @@
-version: ~> 1.0
-
-services:
-  - docker
```
Comment (gr2m) [inline .travis.yml:4]: what's the equivalent for this in GitHub Actions? 🤔

**Comment by HonkingGoose** [comment]
There's a spelling error in your PR title. You might want to fix that. :wink: 

> ci: **repalce** travis with actions, enable renovate

**Comment by gr2m** [comment]
thanks @HonkingGoose 

this PR is ready to review, let me know if anything is missing or looks wrong. Once we merge this I'll do the same PR for the remaining `semantic-release/*` repositories

File: .travis.yml (hunk 1 above)
Comment (HonkingGoose) [reply .travis.yml:4]: I don't know if you've seen this migration guide that GitHub themselves wrote to migrate from Travis -> GitHub Actions:
https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions

---

I think I found something relevant here:

https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-nodejs#running-on-a-different-operating-system which says:

> You can also run jobs in Docker containers, or you can provide a self-hosted runner that runs on your own infrastructure. For more information, see "Workflow syntax for GitHub Actions."

That quoted section links to this: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on if you search for the string `docker` on this page, you'll get 51 hits, maybe something in that page can help you get further.

I'm not going to recommend any course of action, as I'm not qualified to do so... :smile:

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: ubuntu-latest
+    needs: test_matrix
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: bahmutov/npm-install@v1
+      - run: npm run lint
```
Comment (HonkingGoose) [inline .github/workflows/test.yml:50]: I think you're missing a fresh newline at the end of line 50?

File: .travis.yml (hunk 1 above)
Comment (gr2m) [reply .travis.yml:4]: thanks for digging this up! I've looked through it but didn't find anything that would setup the `docker` binary. For now we will run the tests only on ubuntu which seems to have it installed

File: .github/workflows/test.yml (hunk 2 above)
Comment (gr2m) [reply .github/workflows/test.yml:50]: cheers!

**Review by travi — APPROVED** [review]

**Pull request merged**